### PR TITLE
Disable telemetry (for now) in Sourcegraph App

### PR DESCRIPTION
**Pull request by marekweb**
Disable telemetry ("update check" aka "pings") in Sourcegraph App for now.

The second part of this plan is to re-enable telemetry for opt-in users (where opt-in may be defined as "connected a Sourcegraph.com user")

## Test plan

- Run in App mode
- Check that no update check is being sent

### CODE REVIEW

File: cmd/frontend/internal/app/updatecheck/client.go [hunk 1]
```diff
@@ -743,6 +743,14 @@ func check(logger log.Logger, db database.DB) {
 	mu.Unlock()
 }
 
+// Check if update check is enabled, running at every interval so that it can be
+// switched on/off at runtime.
+func isUpdateCheckCurrentlyEnabled() bool {
+	// Disabled when running in App mode.
+	// TODO: re-enable in App mode when authenticated.
```
Comment (slimsag) [inline cmd/frontend/internal/app/updatecheck/client.go:750]:
```suggestion
	// TODO(app): re-enable in App mode when a Sourcegraph.com account is connected.
```

**Review by slimsag — APPROVED** [review]
LGTM but wait to merge this until we hear a final decision on this. I'll let you know.

**Comment by marekweb** [comment]
@slimsag Sounds good. I figured I'd take a stab at this now but I won't merge until we confirm the plan.

**Comment by marekweb** [comment]
This is now superseded by https://github.com/sourcegraph/sourcegraph/pull/48962

**Pull request closed without merging**